### ISSUE TITLE
Option to delete local file when item is completed

### ISSUE THREAD

**Issue by revilo951** [advplyr/audiobookshelf-app#293]
### Is your feature request related to a problem? Please describe.
Having to manually delete items that are finished.

### Describe the solution you'd like
A local setting in the app that automatically deletes the file(s) when the item is completed.

### Describe alternatives you've considered
Could also be made a server-side setting so the local files being removed are "synced" across devices, ie if I have a podcast downloaded on my tablet and phone, and i finish it on my phone, the next time my tablet syncs with the server it deletes it's own local copy of that file too.

**Comment by hellofaduck** [comment]
It's very good idea, i think about it just today. I've listeting now a big collection of little novels instead of full-sized books, because of this i need to download and then delete 2-3-4 files at day and i found this a little annoying when you doing this on day to day basis, it's not a poblem but it's stupid :) i think autodeletion of already readed books is very obvious feature, my vote for this thing!

**Comment by NeskireDK** [comment]
I really like the suggestion of auto deleting once completed. 
Its generally a pain deleting books atm.

**Comment by advplyr** [comment]
Automatically deleting the items downloaded on the mobile app is a feature I think we can add.

I'm not sure about automatically deleting podcast episodes from the server on completion because then we have to tie those episodes to a specific user. If you have a server with multiple users we would have to know which user completed state matters with this episode. We could have logic that it only applies to the root user but this doesn't feel like a clean solution.

**Comment by advplyr** [comment]
Related to #175

**Comment by revilo951** [comment]
I'm definitely talking about only removing from mobile devices (limited storage etc). If you have more than one user, removing items from the server behind more complicated. (Maybe it could be done once all users have marked an episode as finished, but I like to keep all my episodes as a bit of an archive, in case the podcast goes away)

**Comment by Mega-Volti** [comment]
Definitely a great idea. Not only for mobile devices, also as a per-library setting server-side.

I wouldn't want to delete my books on my server after listening to them, but automatically deleting podcasts on the server after listening would save a lot of manual steps. With a server-side library setting, it would be possible to enable this for the podcasts library but keep it turned off for the book library.

**Comment by tonyedwardspz** [comment]
Being able to delete completed local podcasts in bulk would by useful for the usage pattern described in #958. Deleting each episode individually is cumbersome, and deleting the entire local series isn't ideal.

Rather than / along side automatically deleting listed to episodes as desbribed, a button/option somewhere in the app to delete completed episodes would simplify managaing offline first podcast listening. 

This would fit well into the same menu as Manage Local Files & More info on a podcast series page. Alternativly, it could be on the manage local files views, as another option next to "Delete Local Item".

**Comment by snonux** [comment]
Also, deleting automatically on the server, would be great. But that will only work when all users have completed it, i guess.

**Comment by fede843** [comment]
I would be happily using this feature if available!

**Comment by mathmaniac43** [comment]
Deleting on the server could be helpful. Perhaps users could "subscribe" to podcasts that they want to ensure are stored on the server until they have a chance to listen. An admin user could control the settings for whether episodes are kept indefinitely, or pruned once all "subscribers" have had a chance to complete the podcast.

**Comment by jurgenhaas** [comment]
I don't think this is about deleting episodes from the server. This issue queue is about the app, and I really wanted to have that feature as well, that if I had downloaded episodes locally, that they should be deleted locally when I've completed them. Not on the server, though.

**Comment by cutiepoka** [comment]
I am currently working on a PR for this. If I completed this would it have a chance to get approved @advplyr? My current logic flow is to start up a background task and run that background task every day. Check if the item is finished more than 1 day ago, then delete it if it is. If Not I might just do something more like #1043 and just delete everything finished as that is easier

**Comment by jay-oswald** [comment]
@cutiepoka if that could be a configurable option that would be great, can see plenty of people wanting it asap, and others wanting a delay

**Comment by Bishop-trevorstuart** [comment]
I hope to see this in the android app.

**Comment by commodore-84** [comment]
This has been open since 2022 | what is the likelihood of this coming to fruition?
I appreciate keeping books - but podcasts can just grow and grow.
Ideal solution = delete after listen to option
Acceptable solution = weekly purge job?
